### PR TITLE
Added V1.3 interfaces

### PR DESCRIPTION
**Pull request by vbenni**
Added  support for V1.3 interfaces.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+*.orig
```
Comment (anikulk) [inline .gitignore:1]: Has this file been pushed by mistake?

File: BasePreparedModel.cpp [hunk 2]
```diff
@@ -375,6 +401,21 @@ Return<ErrorStatus> BasePreparedModel::execute_1_2(const Request& request, Measu
     return executeBase(request, measure, this, callback);
 }
 
+Return<V1_3::ErrorStatus> BasePreparedModel::execute_1_3(const V1_3::Request& request, V1_2::MeasureTiming measure,
+                                                  const V1_3::OptionalTimePoint& deadline, const V1_3::OptionalTimeoutDuration& loopTimeoutDuration,
+                                                  const sp<V1_3::IExecutionCallback>& callback) {
+    ALOGV("Entering %s", __func__);
+    return convertToV1_3(executeBase(convertToV1_0(request), measure, this, callback));
+    //return V1_3::ErrorStatus::NONE;
```
Comment (anikulk) [inline BasePreparedModel.cpp:409]: Commented line

File: Driver.cpp [hunk 3]
```diff
@@ -24,6 +25,7 @@
 #include "ModelManager.h"
 #include "ValidateHal.h"
 
+
```
Comment (anikulk) [inline Driver.cpp:28]: Unnecessary empty line

File: Driver.cpp [hunk 4]
```diff
@@ -245,6 +243,171 @@ Return<ErrorStatus> Driver::prepareModelFromCache(
...
+    ALOGV("Exiting %s", __func__);
+    return V1_3::ErrorStatus::GENERAL_FAILURE;
+}
+
+Return<void> Driver::allocate(const V1_3::BufferDesc& desc,
+                        const hidl_vec<sp<V1_3::IPreparedModel>>& preparedModels,
+                        const hidl_vec<V1_3::BufferRole>& inputRoles,
+                        const hidl_vec<V1_3::BufferRole>& outputRoles,
+                        V1_3::IDevice::allocate_cb cb) {
+    ALOGV("hal_1_3::ArmnnDriver::allocate()");
```
Comment (anikulk) [inline Driver.cpp:382]: looks like a copy paste error?

File: ModelManager.h [hunk 5]
```diff
@@ -36,21 +36,21 @@ class NnapiModelInfo {
...
-    size_t getModelOutputsSize() { return mModel.outputIndexes.size(); }
+    size_t getModelOutputsSize() { return mModel.main.outputIndexes.size(); }
 
     // Index into the operand vector
-    OperandLifeTime getOperandLifetime(uint32_t operandIdx) {
-        auto tmpOperand = mModel.operands[operandIdx];
-        return tmpOperand.lifetime;
+    V1_3::OperandLifeTime getOperandLifetime(uint32_t operandIdx) {
+        auto tmpOperand = mModel.main.operands[operandIdx];
+        return (tmpOperand.lifetime);
```
Comment (anikulk) [inline ModelManager.h:50]: Coding style "( )" not needed

**Review by anikulk — COMMENTED** [review]
I see many files which are empty. IIUC executeFenced & Burst are not supported, so VTS will fail? Is this right? Can we capture that Burst & fenced are not supported yet in the commit msg.

File: .gitignore (hunk 1 above)
Comment (vbenni) [reply .gitignore:1]: yes.. will remove it

File: ModelManager.h (hunk 5 above)
Comment (vbenni) [reply ModelManager.h:50]: Noted

**Comment by vbenni** [comment]
> I see many files which are empty. IIUC executeFenced & Burst are not supported, so VTS will fail? Is this right? Can we capture that Burst & fenced are not supported yet in the commit msg.

executeFenced is not yet supported.
I believe burst is only supported in v1.2. Need to cross verify. @RavirajSitaram , correct if I am wrong.

**Comment by vbenni** [comment]
Raising a new PR.

**Pull request closed without merging**